core/domain/stats_services.py

```diff
@@ -47,36 +48,65 @@ def get_exps_unresolved_answers_count_for_default_rule(exp_ids):
     Note that this method currently returns the counts only for the DEFAULT
     rule. This should ideally handle all types of unresolved answers.
 
-    Returns:
-        A dict, keyed by the string '{exp_id}', whose values are the number of
-        unresolved answers that exploration has.
+    Returns a dict of the following format:
+        {
+          'exp_id_1': {
+            'count': (number of unresolved answers for this exploration),
+            'values': [
+              {'count': 4, value: 'answer_1'},
+              {'count': 2, value: 'answer_2'},
+              {'count': 1, value: 'answer_3'}
+            ]
+          },
+          'exp_id_2': {
+            'count': (number of unresolved answers for this exploration),
+            'values': [
+              {'count': 8, value: 'answer_4'},
+              {'count': 3, value: 'answer_5'},
+              {'count': 1, value: 'answer_6'}
+            ]
+          }
+        }
     """
-    explorations = exp_services.get_multiple_explorations_by_id(exp_ids)
-
-    # The variable `exploration_states_tuples` is a list of all
-    # (exp_id, state_name) tuples for the given exp_ids.
-    # E.g. - [
-    #   ('eid1', 'Introduction'),
-    #   ('eid1', 'End'),
-    #   ('eid2', 'Introduction'),
-    #   ('eid3', 'Introduction')
-    # ]
-    # when exp_ids = ['eid1', 'eid2', 'eid3'].
-    explorations_states_tuples = [
-        (exp_domain_object.id, state_key)
-        for exp_domain_object in explorations.values()
-        for state_key in exp_domain_object.states
-    ]
+    def _get_explorations_state_tuples_by_ids(exp_ids):
```

Comment:
Yeah, is used just once. That's why took it out as a function within this function itself.
Makes the code more readable IMO. Will be happy to revert it back, if you think otherwise.